### PR TITLE
Expose remaining functions and types from `cardano-ledger-*` required by `cardano-cli`

### PR DESCRIPTION
**Pull request by palas**
# Changelog

```yaml
- description: |
    Expose transitive dependencies from `cardano-ledger` needed by `cardano-cli`
    Breaking change: `Cardano.Api.Ledger` no longer exports Byron's `Tx` but it is now exported in `Cardano.Api.Byron`
  type:
  - breaking       # the API has changed in a breaking way
```

# Context

As part of the work targeted at addressing issues pointed out in https://github.com/IntersectMBO/cardano-api/issues/608, this PR aims to provide `cardano-cli` with everything it needs from ledger.

# How to trust this PR

Make sure it is just a refactoring.

Look at it in conjunction with: https://github.com/IntersectMBO/cardano-cli/pull/941

Take advantage that there are one commit for `shelley`, one for `core`, and one for `api` (same for the corresponding PR in cardano-cli)

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Byron.hs [hunk 1]
```diff
@@ -154,14 +155,14 @@ module Cardano.Api.Byron
   )
 where
 
-import           Cardano.Api hiding (Address, Certificate, Lovelace, NetworkMagic, TxIn, TxOut,
-                   UTxO (..))
+import           Cardano.Api hiding (Address, Certificate, Lovelace, NetworkMagic, Tx (..), TxIn,
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Byron.hs:158]: Not for this PR but we shouldn't be importing `Cardano.Api` here. We need to import the types from their relevant internal modules.

File: cardano-api/internal/Cardano/Api/ReexposeLedger.hs [hunk 2]
```diff
@@ -88,7 +94,6 @@ module Cardano.Api.ReexposeLedger
   -- Byron
   , Annotated (..)
   , byronProtVer
-  , Byron.Tx (..)
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/ReexposeLedger.hs:91]: Why is this removed?

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/ReexposeLedger.hs (hunk 2 above)
Comment (palas) [reply cardano-api/internal/Cardano/Api/ReexposeLedger.hs:91]: Because it clashes with the `Tx` from `Cardano.Ledger.Alonzo.Core` that I had to re-export for removing `cardano-ledger-api` dependency. On the other hand, it seemed to make sense because of what you said that Byron stuff should be in `Cardano.Api.Byron`. And [here](https://github.com/IntersectMBO/cardano-cli/pull/941/commits/c35228407cd7de8685ca693f2d153dedf618ed1e#diff-87ab989fc0540162119ac2e0e294e56a86fdb9abe26cfdc5a0b27e19849ba5b3R129), for example, it was the only thing that wasn't being imported from `Cardano.Api.Byron`, so it seems more consistent.
